src/plugins/interactive_setup/public/progress_indicator.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { EuiStepProps } from '@elastic/eui';
+import { EuiCallOut, EuiPanel, EuiSteps } from '@elastic/eui';
+import type { FunctionComponent } from 'react';
+import React, { useEffect } from 'react';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+import useTimeout from 'react-use/lib/useTimeout';
+import useTimeoutFn from 'react-use/lib/useTimeoutFn';
+
+import { i18n } from '@kbn/i18n';
+
+import { useHttp } from './use_http';
+
+export interface ProgressIndicatorProps {
+  onSuccess?(): void;
+}
+
+export const ProgressIndicator: FunctionComponent<ProgressIndicatorProps> = ({ onSuccess }) => {
+  const http = useHttp();
+  const [status, checkStatus] = useAsyncFn(async () => {
+    let isAvailable: boolean | undefined = false;
+    let isPastPreboot: boolean | undefined = false;
+    try {
+      const { response } = await http.get('/api/status', { asResponse: true });
+      isAvailable = response ? response.status < 500 : undefined;
+      isPastPreboot = response?.headers.get('content-type')?.includes('application/json');
+    } catch ({ response }) {
+      isAvailable = response ? response.status < 500 : undefined;
+      isPastPreboot = response?.headers.get('content-type')?.includes('application/json');
+    }
```

Comment:
__question:__ would you mind explaining the idea behind these checks?

Reply:
In total it takes roughly 20 seconds to go from preboot to Kibana being operation. This is too long to expect users to stare at a loading spinner. In order to be able to show some kind of progress we perform these checks to infer what stage Kibana is at. It's not ideal but it's the best I could come up with in the absence of interactive setup getting notified by core how far down the line it is. 

Do you have another idea how we could track startup progress?